### PR TITLE
service-hacluster-remote: initial commit

### PR DESCRIPTION
**Pull request by grueni**

### CODE REVIEW

File: components/cluster/service-hacluster-remote/service-hacluster-remote.license [hunk 1]
```diff
@@ -0,0 +1,2 @@
+TBD
```
Comment (xen0l) [inline components/cluster/service-hacluster-remote/service-hacluster-remote.license]: We could use a license text here.

**Review by xen0l — CHANGES_REQUESTED** [review]

**Review by xen0l — APPROVED** [review]

File: components/cluster/service-hacluster-remote/files/hacluster-remote.xml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      <service_fmri value='svc:/system/filesystem/local:default'/>
+    </dependency>
+    <dependency name='autofs' grouping='optional_all' restart_on='error' type='service'>
+      <service_fmri value='svc:/system/filesystem/autofs:default'/>
+    </dependency>
+    <method_context>
+      <method_credential user='root'/>
+    </method_context>
+    <exec_method name='start' type='method' exec='/lib/svc/method/svc-hacluster-remote start' timeout_seconds='60'/>
+    <exec_method name='stop' type='method' exec='/lib/svc/method/svc-hacluster-remote stop' timeout_seconds='60'/>
```
Comment (xen0l) [inline components/cluster/service-hacluster-remote/files/hacluster-remote.xml]: :kill is preferred here

File: components/cluster/service-hacluster-remote/files/svc-hacluster-remote [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+killapp() {
+        pid=`pgrep -f $1`
+        if [ "x${pid}" != "x" ]; then
+                kill -9 ${pid}
+        fi
+        return 0
+}
+
+start() {
+        stop
```
Comment (xen0l) [inline components/cluster/service-hacluster-remote/files/svc-hacluster-remote]: Drop this as SMF has :kill builtin and that has better results. Also, doing pkill et al from SMF manifest is not the best practice.

**Review by xen0l — CHANGES_REQUESTED** [review]

File: components/cluster/service-hacluster-remote/Makefile [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+# If applicable, add the following below this CDDL HEADER, with the
+# fields enclosed by brackets "[]" replaced with your own identifying
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+#
+
+#
+# Copyright 2016 Andreas Grueninger, Grueninger GmbH, (grueni). All rights reserved.
```
Comment (xen0l) [inline components/cluster/service-hacluster-remote/Makefile]: Bump the year here, please.

File: components/cluster/service-hacluster-remote/files/svc-hacluster-remote [hunk 5]
```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/bash
```
Comment (xen0l) [inline components/cluster/service-hacluster-remote/files/svc-hacluster-remote]: Why bash and not /bin/ksh, like everywhere?

File: components/cluster/service-hacluster-remote/files/svc-hacluster-remote [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+# PCMK_trace_formats=1
+# PCMK_trace_tags=1
+
+export PCMK_ipc_type=socket
+export PCMK_logfile=/var/log/hacluster/pacemaker.log
+
+export CLUSTER_USER=hacluster
+export CLUSTER_GROUP=haclient
+export PATH=/usr/gnu/bin:${PATH}
+MYNAME=hacluster
```
Comment (xen0l) [inline components/cluster/service-hacluster-remote/files/svc-hacluster-remote:25]: Why not also export these?

File: components/cluster/service-hacluster-remote/files/svc-hacluster-remote [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+           ;;
+        'restart')
+           stop
+           start
+           ;;
+        'stop')
+           stop
+           ;;
+        *)
+           echo "Usage: -bash { start | stop | restart}"
```
Comment (xen0l) [inline components/cluster/service-hacluster-remote/files/svc-hacluster-remote]: -bash -> $0

File: components/cluster/service-hacluster-remote/files/svc-hacluster-remote (hunk 6 above)
Comment (grueni) [reply components/cluster/service-hacluster-remote/files/svc-hacluster-remote:25]: This and the following variables are only used in the shell script and not by any subprocess

**Review by xen0l — APPROVED** [review]

**Pull request merged**